pennylane/measurements/measurements.py

```diff
@@ -331,9 +330,11 @@ def eigvals(self):
         Returns:
             array: eigvals representation
         """
+        if self.obs.__class__.__name__ == "MeasurementValue":
+            return qml.math.arange(0, 2 ** len(self.wires), 1)
+
         if self.obs is not None:
-            with contextlib.suppress(qml.operation.EigvalsUndefinedError):
-                return self.obs.eigvals()
+            return self.obs.eigvals()
```

Comment:
Somewhat of a breaking change, tagging @trbromley for feedback.

Reply:
Yep, that's about it. If `mp.eigvals()` were requested for a measurement process with an observable with no eigenvalues, previously we would suppress the error and return `mp._eigvals`, which is `None` unless changed manually (which is shouldn't be).

I think this change makes sense because returning `None` will lead to errors down the line, but the errors will not be descriptive.

Reply:
Thanks! I think this is fine. Please can you add an entry to the breaking changes part of the changelog.